api/v1/egressgateway_types.go

```diff
@@ -0,0 +1,277 @@
+	// Default: 5
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=2147483647
+	// +optional
+	IntervalSeconds *int32 `json:"intervalSeconds"`
+
+	// TimeoutSeconds defines the timeout value of ICMP probes. Used when IPs is non-empty.
+	// Default: 15
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=2147483647
+	// +optional
+	TimeoutSeconds *int32 `json:"timeoutSeconds"`
+}
+
+// HTTPProbe defines the HTTP probe configuration for Egress Gateway.
+type HTTPProbe struct {
+	// URLs define the list of HTTP probe URLs. Egress Gateway will probe each URL
+	// periodically.If all probes fail, Egress Gateway will report non-ready.
+	// +optional
+	URLs []string `json:"urls,omitempty"`
+
+	// IntervalSeconds defines the interval of HTTP probes. Used when URLs is non-empty.
+	// Default: 10
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Maximum=2147483647
+	// +optional
+	IntervalSeconds *int32 `json:"intervalSeconds"`
+
+	// TimeoutSeconds defines the timeout value of HTTP probes. Used when URLs is non-empty.
+	// Default: 30
+	// +optional
+	TimeoutSeconds *int32 `json:"timeoutSeconds"`
+}
+
+// AWSEgressGateway defines the configurations for deploying EgressGateway in AWS
+type AWSEgressGateway struct {
+
+	// NativeIP defines if EgressGateway is to use an AWS backed IPPool.
+	// Default: Disabled
+	// +kubebuilder:validation:Enum=Enabled;Disabled
+	// +optional
+	NativeIP *NativeIP `json:"nativeIP,omitempty"`
+
+	// ElasticIPs defines the set of elastic IPs that can be used for Egress Gateway pods.
+	// NativeIP must be Enabled if elastic IPs are set.
+	// +optional
+	ElasticIPs []string `json:"elasticIPs,omitempty"`
+}
+
+// EgressGatewayStatus defines the observed state of EgressGateway
+type EgressGatewayStatus struct {
+	// State provides user-readable status.
+	State string `json:"state,omitempty"`
+
+	// Conditions represents the latest observed set of conditions for the component. A component may be one or more of
+	// Ready, Progressing, Degraded or other customer types.
+	// +optional
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+
+// EgressGateway is the Schema for the egressgateways API
+type EgressGateway struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   EgressGatewaySpec   `json:"spec,omitempty"`
+	Status EgressGatewayStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+
+// EgressGatewayList contains a list of EgressGateway
+type EgressGatewayList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []EgressGateway `json:"items"`
+}
+
+func (c *EgressGateway) GetLogSeverity() string {
+	return string(*c.Spec.LogSeverity)
+}
+
+func (c *EgressGateway) GetTerminationGracePeriod() *int64 {
```

Comment:
I think you want to do nil checking on these, right? 

See typha as an example: https://github.com/tigera/operator/blob/74f866faa72aa7be30583b207c7a05fc26540809/api/v1/typha_deployment_types.go#L269-L275

Reply:
I read that  default value will be used  ```TerminationGracePeriodSeconds``` is ```nil```. Hence I thought I need not do a nil check.

Reply:
It's not a matter of default values, it's to avoid a nil pointer dereference in case any of those intermediate structures is nil.